bindings/java/src/org/sleuthkit/datamodel/blackboardutils/CommunicationArtifactsHelper.java

```diff
@@ -251,17 +277,58 @@ public BlackboardArtifact addContact(Account.Address contactAddress,
 		attributes.addAll(additionalAttributes);
 		contactArtifact.addAttributes(attributes);
 
-		// Find/Create an account instance for the contact
-		// Create a relationship between selfAccount and contactAccount
-		AccountFileInstance contactAccountInstance = createAccountInstance(accountsType, contactAddress.getUniqueID());
-		addRelationship(selfAccountInstance, contactAccountInstance, contactArtifact, Relationship.Type.CONTACT, 0);
-
+		// create an account for each specified contact method, and a relationship with self account
+		createContactMethodAccountAndRelationship(Account.Type.PHONE, phoneNumber, contactArtifact, 0);
+		createContactMethodAccountAndRelationship(Account.Type.PHONE, homePhoneNumber, contactArtifact, 0);
+		createContactMethodAccountAndRelationship(Account.Type.PHONE, mobilePhoneNumber, contactArtifact, 0);
+		createContactMethodAccountAndRelationship(Account.Type.EMAIL, emailAddr, contactArtifact, 0);
+
+		// if the additional attribute list has any phone/email/id attributes, create accounts & relationships for those. 
+		if ((additionalAttributes != null) && hasAnyIdAttribute) {
```

Comment:
Why must an ID attribute be required here? If I add contact name and email (valid) and my additional attributes list is all TSK_PHONE then they will not be added.